CMakeLists.txt

```
@@ -274,6 +274,16 @@ endif(BUILD_WITH_AUTODIFF_SUPPORT)
 
 if(BUILD_WITH_CASADI_SUPPORT)
   add_project_dependency(casadi 3.4.5 REQUIRED PKG_CONFIG_REQUIRES "casadi >= 3.4.5")
+
+  # Normalize CasADi link target across versions. Prefer imported target (casadi::casadi), then
+  # legacy target (casadi)
+  if(TARGET casadi::casadi)
+    set(CASADI_TARGET casadi::casadi)
+  elseif(TARGET casadi)
+    set(CASADI_TARGET casadi)
```

```suggestion
    add_library(casadi::casadi ALIAS casadi)
```

(this is only a suggestion, the current solution would also be fine for me, but this seems easier to maintain on the long term)

---

You're right.
I was thinking CMake would prohibit to create a library with a different namespace than our project. But a local test prove me wrong.